lib/rules/no-underscore-dangle.js

```diff
@@ -126,16 +131,40 @@ module.exports = {
         }
 
         /**
-         * Check if function has a underscore at the end
+         * Check if function parameter has a dangling underscore.
+         * @param {ASTNode} node node to evaluate
+         * @returns {boolean} true if function parameter name has a dangling underscore.
+         * @private
+         */
+        function checkForDanglingUnderscoreInFunctionParameter(node) {
+            if (!allowFunctionParams) {
+                node.params.forEach(param => {
+                    const identifier = param.name;
+
+                    if (typeof identifier !== "undefined" && hasDanglingUnderscore(identifier) && !isAllowed(identifier)) {
+                        context.report({
+                            node,
+                            messageId: "unexpectedUnderscore",
+                            data: {
+                                identifier
+                            }
+                        });
+                    }
+                });
+            }
+        }
+
+        /**
+         * Check if function has a dangling underscore
          * @param {ASTNode} node node to evaluate
          * @returns {void}
          * @private
          */
-        function checkForTrailingUnderscoreInFunctionDeclaration(node) {
+        function checkForDanglingUnderscoreInFunction(node) {
             if (node.id) {
```

Comment:
```suggestion
            if (node.type === "FunctionDeclaration" && node.id) {
```

@sunghyunjo it seems there's a consensus to revert the behavior for `FunctionExpression#id` (in other words, to not check that identifier), so we could add here a condition to check only function declarations.

It would be nice to also add a `valid` regression test with a function expression that has a dangling underscore in its name.